lib/kafka_producer.js

```diff
@@ -209,8 +212,15 @@ KafkaProducer.prototype.batch = function batch(topic, message, timeStamp, callba
         return;
     }
 
+    if (typeof message !== 'string' && !Buffer.isBuffer(message)) {
+        var err = new Error('For batching, message must be a string or buffer, not ' + (typeof message));
+        if (callback) {
+            callback(err);
+        }
+        return;
+    }
```

Comment:
For consistency, make it like:
if (typeof message !== 'string' && !Buffer.isBuffer(message)) {
        if (callback) {
            callback(new Error('For batching, message must be a string or buffer, not ' + (typeof message)));
        } // TODO: else { self.logger.error() }
        return;
    }